### PR TITLE
EQL: Adds "fields" request field to the eql request (#68962)

### PR DESCRIPTION
**Pull request by astefan**
Backport of https://github.com/elastic/elasticsearch/pull/68962.
(cherry picked from commit 22c880e641ef36df67a7a616de0c4a25e48d9785)

### CODE REVIEW

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchResponseTests.java [hunk 1]
```diff
@@ -78,7 +101,14 @@ public BytesReference toBytes(XContentType type) {
             hits = new ArrayList<>();
             for (int i = 0; i < size; i++) {
                 BytesReference bytes = new RandomSource(() -> randomAlphaOfLength(10)).toBytes(xType);
-                hits.add(new Event(String.valueOf(i), randomAlphaOfLength(10), bytes));
+                Map<String, DocumentField> fetchFields = new HashMap<>();
+                for (int j = 0; j < randomIntBetween(0, 5); j++) {
```
Comment (matriv) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchResponseTests.java:105]: Shouldn't this be a var before the loop?

File: x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/10_basic.yml [hunk 2]
```diff
@@ -49,6 +59,55 @@ setup:
   - match: {hits.events.1._id: "2"}
   - match: {hits.events.2._id: "3"}
 
+---
+"Execute EQL events query with fields filtering":
+  - do:
```
Comment (matriv) [inline x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/10_basic.yml:64]: should those have maybe the:
```
- skip:
      version: " - 7.12.99"
```
block?

**Review by matriv — APPROVED** [review]
LGTM. Left 2 comments.

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchResponseTests.java [hunk 3]
```diff
@@ -217,5 +217,36 @@ public static EqlSearchResponse createRandomInstance(TotalHits totalHits, XConte
...
+        if (original == null || original.isEmpty()) {
+            return original;
+        }
+        List<Event> mutatedEvents = new ArrayList<>(original.size());
+        for(Event e : original) {
+            if (version.onOrAfter(Version.V_7_13_0)) {
+                mutatedEvents.add(new Event(e.index(), e.id(), e.source(), e.fetchFields()));
+            } else {
+                mutatedEvents.add(new Event(e.index(), e.id(), e.source(), null));
+            }
```
Comment (bpintea) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchResponseTests.java:248]: nit: optionally, you could add one line (or two) only, if you condition just last parameter's value by the version (like done in the Request test).

**Review by bpintea — APPROVED** [review]
LGTM.

File: x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/10_basic.yml (hunk 2 above)
Comment (bpintea) [reply x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/10_basic.yml:64]: I guess the 7.12 port will have the versions kept as in original PR?

Comment (astefan) [reply x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/10_basic.yml:64]: There won't be a 7.12 port, only 7.x.

Comment (astefan) [reply x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/10_basic.yml:64]: I don't think the EQL yaml tests run in a mixed cluster (or any other bwc context) scenario that warrants for this `skip` block.

**Comment by astefan** [comment]
@elasticmachine update branch

**Pull request merged**